lib/index.js

```diff
@@ -83,73 +74,76 @@ class Import {
+  let useInput = false,
+    extensions = null;
   let roots = strArray(options);
   if (!roots && options && typeof options === 'object') {
     roots = strArray(options.root);
     extensions = strArray(options.extensions);
 
-    if (~['prepend', 'append'].indexOf(options.useEntry)) {
-      useEntry = options.useEntry;
+    if (['prepend', 'append'].includes(options.useInput)) {
+      useInput = options.useInput;
     }
   }
 
   if (!extensions) extensions = [''];
-  else if (extensions.indexOf('') === -1) extensions.unshift('');
+  else if (!extensions.includes('')) extensions.unshift('');
 
-  // We can cache resolved modules when they're relative to the root. We use a
-  // null-inherited to avoid needing to use hasOwnProperty.
+  // We can cache resolved modules when they're relative to the root. We use a null-inherited to
+  // avoid needing to use hasOwnProperty.
   const cache = Object.create(null);
 
-  const imports = [], hadNoRoots = !roots;
+  const imports = [],
+    hadNoRoots = !roots;
   return {
-    options({entry}) {
+    options({ input }) {
       if (!roots) roots = [];
 
-      // Ignore exotic entry types (arrays, objects) as may be handled by plugins
-      // like https://github.com/rollup/rollup-plugin-multi-entry.
-      if (typeof entry === 'string') {
-        const entryRoot = path.dirname(entry);
+      // Ignore exotic input types (arrays, objects) as may be handled by plugins like
+      // https://github.com/rollup/rollup-plugin-multi-entry.
+      if (typeof input === 'string') {
+        const entryRoot = path.dirname(input);
 
         if (!roots.length) {
           roots.push(entryRoot);
-        } else if (useEntry === 'prepend') {
+        } else if (useInput === 'prepend') {
           roots.unshift(entryRoot);
-        } else if (useEntry === 'append') {
+        } else if (useInput === 'append') {
           roots.push(entryRoot);
         }
       }
 
-      for (let root of roots) {
-        for (let ext of extensions) {
-          imports.push(new Import(root, ext));
+      for (const rootPath of roots) {
+        for (const ext of extensions) {
+          imports.push(new Import(rootPath, ext));
         }
       }
     },
 
     resolveId(importee, importer) {
       // Don't try to resolve the entry unless we have explicit roots.
-      if (!importer && hadNoRoots) return;
+      if (!importer && hadNoRoots) return null;
 
       if (importee[0] === '/') {
-        // If we've cached this import, don't bother trawling the filesystem
-        // again.
+        // If we've cached this import, don't bother trawling the filesystem again.
         const cached = cache[importee];
         if (cached) return cached;
 
-        return resolve(importee, imports).then((resolved) => {
-          // Save this in the cache in case we see it again.
-          cache[importee] = resolved;
-          return resolved;
-        });
+        const resolution = resolve(importee, imports);
```

Comment:
WRT caching, do we have to consider the case where `resolution` rejects?

Reply:
That's a good call, but I think we don't - if it rejects, then everything will come to a halt, and we won't have a poisoned cache.